File: backend/Bingol-University-Social-Media-main/src/main/java/bingol/campus/response/DataResponseMessage.java
package bingol.campus.response;



public class DataResponseMessage<T> extends ResponseMessage {
    private T data;

    public DataResponseMessage(String message, boolean isSuccess, T data) {
        super(message, isSuccess);
        this.data = data;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: frontend/campus_app/lib/models/data_response_message.dart
import 'response_message.dart';

class DataResponseMessage<T> extends ResponseMessage {
  final T? data;

  DataResponseMessage({
    required String message,
    required bool isSuccess,
    required this.data,
  }) : super(message: message, isSuccess: isSuccess);

  factory DataResponseMessage.fromJson(Map<String, dynamic> json, T Function(Object?) fromJsonT) {
    final isSuccess = json['success'] as bool? ?? false;
    
    return DataResponseMessage<T>(
      message: json['message'] as String? ?? '',
      isSuccess: isSuccess,
      data: json['data'] != null ? fromJsonT(json['data']) : null,
    );
  }

  @override
  Map<String, dynamic> toJson() {
    final Map<String, dynamic> json = {
      'message': message,
      'success': isSuccess,
    };
    if (data != null) {
      json['data'] = data;
    }
    return json;
  }
} 